subsystems/drivebase/SwerveDrive.java

```diff
@@ -311,8 +327,24 @@ public void mustangPeriodic() {
         Logger.recordOutput(DRIVEBASE_PITCH, getPitch());
         Logger.recordOutput(DRIVEBASE_ROLL, getRoll());
 
+       
+
+        if (kConfig.kDriveMotorType.equals(Motor_Type.KRAKEN_X60)) {
+            emaCurrent = currentAlpha * ((TalonFX) mModules[0].getDriveMotor()).getSupplyCurrent() + (1 - currentAlpha) * emaCurrent; // weighted current average of front-right drive motor
+            if (emaCurrent > 30) { // TODO tune this? what should be the threshold?
+                currentLimitConfigs.SupplyCurrentLimit = currentSupplyCurrentLimit - 0.25; // TODO what increment to decrease/increase limit? how frequently current readings update?
+            } else {
+                currentLimitConfigs.SupplyCurrentLimit = currentSupplyCurrentLimit + 0.25;
```

Comment:
just go back to the "normal" limit if we think we're now ok. This logic here will increase the supply limit forever if we're sitting and doing nothing. Or doing little. And this is at 50 cycles per second so... that's a lot.